cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -0,0 +1,201 @@
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	"k8s.io/kubernetes/cmd/kubeadm/app/features"
+	kubeletphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	kubeletWriteInitConfigLongDesc = normalizer.LongDesc(`
+		Writes init kubelet configuration to disk for dynamic kubelet configuration feature.
+		Please note that the kubelet configuration can be passed to kubeadm as a value into the master configuration file.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletWriteInitConfigExample = normalizer.Examples(`
+		# Writes init kubelet configuration to disk.
+		kubeadm alpha phase kubelet init
+		`)
+
+	kubeletUploadDynamicConfigLongDesc = normalizer.LongDesc(`
+		Uploads dynamic kubelet configuration as ConfigMap and links it to the current node as ConfigMapRef.
+		Please note that the kubelet configuration can be passed to kubeadm as a value into the master configuration file.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletUploadDynamicConfigExample = normalizer.Examples(`
+		# Uploads dynamic kubelet configuration as ConfigMap.
+		kubeadm alpha phase kubelet upload
+		`)
+
+	kubeletEnableDynamicConfigLongDesc = normalizer.LongDesc(`
+		Enables or updates dynamic kubelet configuration on node. This should be run on nodes.
+		Please note that the kubelet configuration can be passed to kubeadm as a value into the master configuration file.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletEnableDynamicConfigExample = normalizer.Examples(`
+		# Enables dynamic kubelet configuration on node.
+		kubeadm alpha phase kubelet enable
+		`)
+)
+
+// NewCmdKubelet returns main command for Kubelet phase
+func NewCmdKubelet() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "kubelet",
+		Short: "Adopts dynamic kubelet configuration.",
+		Long:  cmdutil.MacroCommandLongDescription,
+	}
+
+	cmd.AddCommand(NewCmdKubeletWriteInitConfig())
+	cmd.AddCommand(NewCmdKubeletUploadDynamicConfig())
+	cmd.AddCommand(NewCmdKubeletEnableDynamicConfig())
+
+	return cmd
+}
+
+// NewCmdKubeletWriteInitConfig calls cobra.Command for writing init kubelet configuration
+func NewCmdKubeletWriteInitConfig() *cobra.Command {
+	var cfgPath string
+	cmd := &cobra.Command{
+		Use:     "init",
+		Short:   "Writes init kubelet configuration to disk",
+		Long:    kubeletWriteInitConfigLongDesc,
+		Example: kubeletWriteInitConfigExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := &kubeadmapiext.MasterConfiguration{
+				// KubernetesVersion is not used by kubelet init, but we set this explicitly to avoid
+				// the lookup of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
+				KubernetesVersion: "v1.9.0",
```

Comment:
I have seen this elsewhere in kubeadm. Let's keep a shared const across all places to facilitate ease of updates. Also should this be v1.10.0 now?

Reply:
Maybe a follow-up issue is better.

Reply:
If it's just a dummy value we should have a dummy const or something